enhancements/network/multi-networkpolicy.md

```diff
@@ -0,0 +1,177 @@
+mechanism to additional network interfaces and it provides customers the ability to filter
+traffic from unexpected sources.
+
+### Goals
+
+- Introduce MultiNetworkPolicy CRD
+- Introduce multi-networkpolicy which generates iptables from CRD as shown in the Design Details section
+- Provide iptable based filtering mechanism for pod
+- Gathering related information (iptables rules in pod network namespace) by must-gather
+
+### Non-Goals
+
+- Provide eBPF filtering mechanism
+- Provide NetworkPolicy for userspace pod which uses DPDK and so on
+
+## Proposal
+
+### User Stories
+
+#### Story 1
+
+Customer can specify how groups of pods are allowed to communicate with each other
+and other network endpoints as Kubernetes NetworkPolicy does.
+
+### Implementation Details/Notes/Constraints
+
+### Multus
+
+No additional change is required because multi-networkpolicy only consumes 
+network-attachment-definition CRD and it does not change net-attach-def CRD.
+
+### How user enables multi-networkpolicy?
+
+Currently we're going to bring it as Tech Preview first and then GA after a few release,
+hence opt-in style configuration is planned at Tech Preview and make it default at GA.
+Opt-in style configuration will be done at cluster-network-operator as boolean flag
+(such as `useMultiNetworkPolicy: true`).
+
+
+### Risks and Mitigations
+
+## Design Details
+
+### MultiNetworkPolicy samples
+
+MultiNetworkPolicy CRD is defined in https://github.com/k8snetworkplumbingwg/multi-networkpolicy/blob/master/scheme.yml
+
+MultiNetworkPolicy CRD has pretty similar schema to Kubernetes NetworkPolicy. The differences are
+'apiVersion' and 'kind', to define it as CRD of MultiNetworkPolicy. Its semantics are same as well
+as Kubernetes other than target interface annotation, `k8s.v1.cni.cncf.io/policy-for`. Target interface
+is specified annotation, `k8s.v1.cni.cncf.io/policy-for`, as net-attach-def name.
+
+```
+apiVersion: k8s.cni.cncf.io/v1beta1
+kind: MultiNetworkPolicy
+metadata:
+  name: policy-test1
+  annotations:
+    k8s.v1.cni.cncf.io/policy-for: macvlan-net1
+spec:
+  podSelector:
+    matchLabels:
+      role: http
+  egress:
+    - to:
+      - podSelector:
+          matchLabels:
+            role: client
+        namespaceSelector:
+          matchLabels:
+            test: default
+    - to:
+      - podSelector:
+          matchLabels:
+            role: not-client
+      ports:
+      - protocol: TCP
+        port: 10080
+      - protocol: TCP
+        port: 20081
+  ingress:
+    - from:
+      - podSelector:
+          matchLabels:
+            role: client
+      ports:
+      - protocol: TCP
+        port: 80
+    - from:
+      - podSelector:
+          matchLabels:
+            role: not-client
+      ports:
+      - protocol: TCP
+        port: 80
+```
+
+### Test Plan
+
+Initially, multi-networkpolicy testing will be done via the upstream repo e2e test
```

Comment:
"upstream repo" is That being https://github.com/k8snetworkplumbingwg/multi-networkpolicy-iptables ? You should mention that somewhere

Reply:
Added upstream repository info.